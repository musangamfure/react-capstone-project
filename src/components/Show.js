import '../styles/Show.css';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';
import { BsArrowRightCircle } from 'react-icons/bs';

function Show({ Show, backgroundColor }) {
  const navigate = useNavigate();

  const handleShowDetails = () => {
    const showId = Show.id;
    navigate(`/ShowDetails/${showId}`);
  };

  return (
    <div className="show-item" style={{ backgroundColor }}>
      <button
        type="button"
        onClick={handleShowDetails}
        aria-label="View more info"
        className="details-btn"
      >
        <BsArrowRightCircle />
      </button>
      <img className="show-image" src={Show.image.medium} alt="" />
      <h3 className="show-name">{Show.name}</h3>
      <h3 className="show-rating">{Show.rating.average}</h3>
    </div>
  );
}

Show.propTypes = {
  Show: PropTypes.shape({
    id: PropTypes.number.isRequired,
    image: PropTypes.shape({
      medium: PropTypes.string.isRequired,
    }).isRequired,
    name: PropTypes.string.isRequired,
    rating: PropTypes.shape({
      average: PropTypes.number.isRequired,
    }).isRequired,
  }).isRequired,
  backgroundColor: PropTypes.string.isRequired,
};

export default Show;
