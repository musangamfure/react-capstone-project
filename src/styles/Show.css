.show-item {
  display: flex;
  flex-direction: column;
  color: var(--light-mode-text);
  align-items: flex-start;
  background-color: var(--color-white);
  margin-bottom: 3rem;
  text-decoration: none;
  box-shadow: 5px 5px 10px 0 #dfd5d5;
}

.show-item,
.view-more,
.details-container {
  background-color: var(--dark-blue-elements);
  color: var(--color-white);
  box-shadow: 5px 5px 10px 0;
  border-radius: 8px;
}

.show-image {
  display: block;
  width: 100%;
  height: 100%;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
}

.show-name,
.view-more {
  margin: 10px 0;
  padding-left: 5px;
}

.details-btn {
  padding: 10px;
  border-radius: 10px;
  font-family: inherit;
  cursor: pointer;
}
