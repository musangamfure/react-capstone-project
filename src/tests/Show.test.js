import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Show from '../components/Show';

test('renders Show component', () => {
  const show = {
    id: 1,
    image: {
      medium: 'abc.jpg',
    },
    name: 'Show Name',
  };
  const backgroundColor = 'rgb(73, 114, 191)';
  const { container } = render(
    <Router>
      <Show Show={show} backgroundColor={backgroundColor} />
    </Router>,
  );

  expect(container.innerHTML).toMatchSnapshot();
});
