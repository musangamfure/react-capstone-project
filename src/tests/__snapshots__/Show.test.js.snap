// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Show component 1`] = `"<div class=\\"show-item\\"><img class=\\"show-image\\" src=\\"abc.jpg\\" alt=\\"\\"><h3 class=\\"show-name\\">Show Name</h3><div class=\\"view-more\\"><button type=\\"button\\" aria-label=\\"View more info\\" class=\\"details-btn\\">View Details</button></div></div>"`;
